packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/PendingRecordingHostApiImpl.java

```diff
@@ -75,7 +75,13 @@ public void handleVideoRecordEvent(@NonNull VideoRecordEvent event) {
     if (event instanceof VideoRecordEvent.Finalize) {
       VideoRecordEvent.Finalize castedEvent = (VideoRecordEvent.Finalize) event;
       if (castedEvent.hasError()) {
-        systemServicesFlutterApi.sendCameraError(castedEvent.getCause().toString(), reply -> {});
+        String cameraErrorMessage;
+        if (castedEvent.getCause() != null) {
+          cameraErrorMessage = castedEvent.getCause().toString();
+        } else {
+          cameraErrorMessage = "Error code " + castedEvent.getError() + ": An error occurred while recording video.";
+        }
+        systemServicesFlutterApi.sendCameraError(cameraErrorMessage, reply -> {});
```

Comment:
The [docs](https://developer.android.com/reference/androidx/camera/video/VideoRecordEvent.Finalize) suggest that if `castedEvent.hasError()` is true, then `castedEvent.getCause()` won't be null, but I've seen `castedEvent.getCause().toString()` throw a `NullPointerException` several times, so I added this logic.